cave/plot/configurator_footprint.py

```diff
@@ -355,131 +371,174 @@ def get_conf_matrix(self):
-            fig.savefig(path)
-
-        # WORK IN PROGRESS
-        # # Show only desired run
-        # handles, labels = ax.get_legend_handles_labels() # return lines and labels
-        # self.logger.debug("Handles: %s", handles)
-        # self.logger.debug("Labels: %s", labels)
-        # interactive_legend = mpld3.plugins.InteractiveLegendPlugin(zip(handles,
-        #                                                          ax.collections),
-        #                                                      labels,
-        #                                                      alpha_unsel=0,
-        #                                                      alpha_over=1,
-        #                                                      start_visible=True)
-        # mpld3.plugins.connect(fig, interactive_legend)
-
-        tooltip = mpld3.plugins.PointHTMLTooltip(scatter, labels,
-                                                 voffset=10, hoffset=10)#, css=self.css)
-
-        mpld3.plugins.connect(fig, tooltip)
-
-        if scatter_inc:
-            tooltip = mpld3.plugins.PointHTMLTooltip(scatter_inc, np.array(labels)[inc_indx].tolist(),
-                                                     voffset=10, hoffset=10)#, css=self.css)
-
-        mpld3.plugins.connect(fig, tooltip)
-
-        if self.output_dir:
-            path = os.path.join(self.output_dir, 'conf_vizs.html')
-            self.logger.debug("Save to %s", path)
-            with open(path, "w") as fp:
-                mpld3.save_html(fig, fp)
-
-        html = mpld3.fig_to_html(fig)
-        plt.close(fig)
-        return html
+            color_mapper = LinearColorMapper(palette="Viridis256",
+                                             low=min_z, high=max_z)
+            p.image(image=contour_data, x=x_range[0], y=y_range[0],
+                    dw=x_range[1] - x_range[0], dh=y_range[1] - y_range[0],
+                    color_mapper=color_mapper)
+            color_bar = ColorBar(color_mapper=color_mapper,
+                                 ticker=BasicTicker(desired_num_ticks=15),
+                                 label_standoff=12,
+                                 border_line_color=None, location=(0,0))
+            p.add_layout(color_bar, 'right')
+
+        # Scatter configurations in order of zorder
+        # TODO expand on multiple configurator runs
+        zorder_values = sorted(list(set(source.data['zorder'])),
+                               key=lambda x: int(x))
+        views = [CDSView(source=source, filters=[
+                            GroupFilter(column_name='zorder', group=z)])
+                 for z in zorder_values]
+        for z, view in zip(zorder_values, views):
+            p.scatter(x='x', y='y',
+                      source=source,
+                      view=view,
+                      color='color', line_color='black',
+                      size='size',
+                      marker=markers[z],
```

Comment:
As you iterate over the values, couldn't you assign the markers here directly?
I.e. first determine which shape the marker should have and then assign it in the scatter method.

Reply:
Pushed a new proposal